cluster/examples/workloads/kubernetes/wordpress/aws/resource_configuration/aws_configuration.sh

```diff
@@ -0,0 +1,212 @@
+  echo "namespace $NAMESPACE doesn't exist"
+  exit -1
+fi
+
+# make sure kubectl is configured
+kubectl cluster-info > /dev/null || echo "KUBECONFIG is not configured properly"
+
+AWS_REGION=$(kubectl -n crossplane-system get provider aws-provider -o=jsonpath='{.spec.region}')
+
+# substitue_variables function accepts a file name, substitutes variable names
+# with their values, and prints out the result to stdout
+# a given file might only have a subset of env variables.
+function substitue_variables {
+  sed \
+    -e "s|((CONFIG_NAME))|"$CONFIG_NAME"|g" \
+    -e "s|((NAMESPACE))|"$NAMESPACE"|g" \
+    -e "s|((AWS_REGION))|"$AWS_REGION"|g" \
+    -e "s|((VPC_ID))|"$VPC_ID"|g" \
+    -e "s|((IG_ID))|"$IG_ID"|g" \
+    -e "s|((SUBNET1_ID))|"$SUBNET1_ID"|g" \
+    -e "s|((SUBNET2_ID))|"$SUBNET2_ID"|g" \
+    -e "s|((SUBNET3_ID))|"$SUBNET3_ID"|g" \
+    -e "s|((RDS_SECURITY_GROUP_ID))|"$RDS_SECURITY_GROUP_ID"|g" \
+    -e "s|((RDS_SUBNET_GROUP_NAME))|"$RDS_SUBNET_GROUP_NAME"|g" \
+    -e "s|((EKS_ROLE_ARN))|"$EKS_ROLE_ARN"|g" \
+    -e "s|((EKS_SECURITY_GROUP_ID))|"$EKS_SECURITY_GROUP_ID"|g" \
+    "$1"
+}
+
+# apply_and_wait_until_ready first substitutes the variabels, then
+# applies the object, and then waits until the resource is ready
+function apply_and_wait_until_ready {
+  echo "applying $1..."
+  k8s_object=$(substitue_variables "$1")
+  echo "$k8s_object" | kubectl apply -f -
+  echo "$k8s_object" | kubectl wait --for=condition=Ready -f - > /dev/null
+  all_objects="${all_objects}
+---
+${k8s_object}"
+}
+
+# delete_resource substitues the variables and then deletes it
+function delete_resource {
+  echo "deleting $1..."
+  substitue_variables "$1" | kubectl delete -f -
+}
+
+function cleanup {
+  echo "cleaning up..."
+
+  # ignore errors in deleting resources, as some resources might already been deleted
+  set +e
+
+  delete_resource rds-securitygroup.yaml
+  delete_resource dbsubnetgroup.yaml
+  delete_resource eks-securitygroup.yaml
+  delete_resource routetable.yaml
+  delete_resource internetgateway.yaml
+  delete_resource subnets.yaml
+  delete_resource iamrole_policies.yaml
+  delete_resource iamrole.yaml
+
+  # retrieve vpc_id to delete orphan dependants. Otherwise it won't get deleted
+  VPC_ID=$(substitue_variables vpc.yaml | kubectl get -o=jsonpath='{.items[0].status.vpcId}' -f -) || echo "cannot retrieve vpcID - skipping deleting vpc related orphan resources"
+  if ! test -z "$VPC_ID"; then
+
+    echo "deleting vpc orphan resources ..."
+    # delete all the load balancers that are externally created in the vpc by k8s
+    vpc_loadbalanancers=($(aws elb describe-load-balancers | jq -rc --arg VPC_ID "$VPC_ID" '.LoadBalancerDescriptions | .[] | select(.VPCId == $VPC_ID) | .LoadBalancerName' ))  
+    for (( i=0; i<${#vpc_loadbalanancers[@]}; i++ )); do
+      echo "deleting load balancer ${vpc_loadbalanancers[i]}"
+      aws elb delete-load-balancer --load-balancer-name ${vpc_loadbalanancers[i]}
+    done
```

Comment:
It sounds like these resources are automatically created when we create an EKS cluster, but not automatically deleted when we delete an EKS cluster. Is this due to a shortcoming in how we're managing EKS clusters, or an inherent property of the EKS API?

Reply:
I think it's the latter. For instance, we you create a service of type `LoadBalancer` in eks, it automatically creates an aws ELB

Reply:
@soorena776 Got it. I think that's true of GCP as well. Is the LB not deleted when the type `LoadBalancer` service is deleted?

Reply:
Correct. I run `kubectl delete app.yaml`, but the ELB stays there still